ingress/Torch-MLIR/generate-mlir.sh

```diff
@@ -33,10 +59,19 @@ source torch-mlir-venv/bin/activate
 # Find script directory
 SCRIPT_DIR=$(dirname "$(readlink -f "$0")")
 
+# Build python arg list
+args=( "--model-entrypoint" "$MODEL" "--dialect" "$DIALECT" )
+[ -n "$STATE_PATH" ]   && args+=( "--model-state-path" "$STATE_PATH" )
+[ -n "$MODEL_ARGS" ]   && args+=( "--model-args" "$MODEL_ARGS" )
+[ -n "$MODEL_KWARGS" ] && args+=( "--model-kwargs" "$MODEL_KWARGS" )
+[ -n "$SAMPLE_SHAPES" ]&& args+=( "--sample-shapes" "$SAMPLE_SHAPES" )
+[ -n "$SAMPLE_FN" ]    && args+=( "--sample-fn" "$SAMPLE_FN" )
+[ -n "$OUT_MLIR" ]     && args+=( "--out-mlir" "$OUT_MLIR" )
+
 # Use the Python script to generate MLIR
-echo "Generating MLIR for model '$MODEL' with dialect '$DIALECT'..."
-python $SCRIPT_DIR/generate-mlir.py --model "$MODEL" --dialect "$DIALECT"
+echo "Generating MLIR for model entrypoint '$MODEL' with dialect '$DIALECT'..."
+python "$SCRIPT_DIR/generate-mlir.py" "${args[@]}"
```

Comment:
Can this entire bash script be folded into the Python script?

At this point I do not seem the .sh giving much value. I guess it necessary for entering the virtualenv, otherwise it's just a wrapper, right?

Reply:
I don't like the idea for a python script to deduce whether it was launched in a proper env and modifying it. I would say a user should be responsible for setting up a proper env before launching the python script (they can always call a bash version of the script that handles venvs for them).

Simplified the `generate-mlir.sh` script so that it only activates venv and forwards all the arguments to the python script.